### DISCUSSION TITLE
Output multiple results / array

### DISCUSSION THREAD

**Discussion by tlwt** [Rooyca/obsidian-api-request#30] [General]
Is it possible to to output multiple rows? I have an api request which return multiple rows which I'd like to output as a table.

**Comment by Rooyca** [comment]
Can you give me an example to see what I can do?

**Comment by tlwt** [comment]
In this example I would like to output a table like this 

| name     |   document_count |
|:---------|-----------------:|
| Angebot  |               17 |
| Auftrag  |               27 |
| Handbuch |               11 |
| Rechnung |              805 |

from the following responds

```
HTTP 200 OK
Allow: GET, POST, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "count": 4,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": 18,
            "slug": "angebot",
            "name": "Angebot",
            "match": "",
            "matching_algorithm": 6,
            "is_insensitive": true,
            "document_count": 17,
            "owner": null,
            "user_can_change": true
        },
        {
            "id": 13,
            "slug": "auftrag",
            "name": "Auftrag",
            "match": "",
            "matching_algorithm": 6,
            "is_insensitive": true,
            "document_count": 27,
            "owner": null,
            "user_can_change": true
        },
        {
            "id": 12,
            "slug": "handbuch",
            "name": "Handbuch",
            "match": "",
            "matching_algorithm": 6,
            "is_insensitive": true,
            "document_count": 11,
            "owner": null,
            "user_can_change": true
        },
        {
            "id": 1,
            "slug": "rechnung",
            "name": "Rechnung",
            "match": "",
            "matching_algorithm": 6,
            "is_insensitive": true,
            "document_count": 805,
            "owner": null,
            "user_can_change": true
        }
    ]
}

```

**Reply by Rooyca** [reply]
I added the `maketable` flag which renders the response as a table. Here is an example (am using `&` to get multiple values from the array)
~~~markdown
```req 
url: http://ws.audioscrobbler.com/2.0/?method=user.getrecenttracks&user=rooyca&api_key=API_KEY&format=json&limit=4
show: recenttracks -> track -> {..} -> name & artist.#text & streamable
maketable: name, artist, stream
```
~~~

**Note that when accessing a nested element we use dot `.` and not `->`**

**Comment by tlwt** [comment]
Overall very nice! Thanks! 

There is a bug, if a comma is in the returned string. I guess you use it as a separator for the table generation.

I get the following results:

id | title
-- | --
9829 | invoice 2023 (133
33 EUR) | 7693
Invoice 2022 | 7688
Invoice 2021 (100 EUR)

It should look like this:

id | title
-- | --
9829 | invoice 2023 (133,33 EUR) 
7693 | Invoice 2022 
7688 | Invoice 2021 (100 EUR)

**Reply by Rooyca** [reply]
Hey @tlwt,
Thanks for pointing that out. It turned out to be a bigger bug affecting every response, not just `maketable`. It should be fixed now.

P.S. Thank you very much for your kind donations, I really appreciate them (I haven't been able to receive even the first one you sent me yet, because GitHub keeps saying that I have to *wait X amount of time*. I hope I can receive everything you have sent me someday. Anyway, thank you very much).

**Reply by tlwt** [reply]
Awesome, that works nicely. 

One additional request: if one column could point to a url that would be helpful.

E.g. in above scenario for me it would be to link the `ID` column to `http://example.com/document/{id}`

**Reply by Rooyca** [reply]
Do you mean passing the response of a second request to a table? Something like this:

~~~markdown
```req
url: http://example.com/
show: name
maketable: name
```
~~~
~~~markdown
```req
url: http://example.com/document/{id}
show: id
maketable: id
```
~~~

Do you want the two responses in a single table? If so, it would be better to use `req-id` to store the responses and then dynamically create a table from those responses using `dataview`:

~~~markdown
```dataview
dv.paragraph(localStorage.getItem("req-ID"))
```
~~~

**Reply by tlwt** [reply]
Sorry for not being clear on this. The use case behind is, that I query my document management system paperless-ngx via API. And I would like to link to the documents shown. I just need a hyperlink to click on with the ID in it so I get directly to the document in a browser.

```req
url: http://example.com/api/documents?api_key=API_KEY...
link: http://example.com/document/{id}
show: results -> {..} -> id & title
maketable: ID, Title
```

should result in a table like

| id |	title | 
| - | --- | 
| [9829](http://example.com/document/9829)  | invoice 2023 (133,33 EUR) |
| [7693](http://example.com/document/7693) | Invoice 2022 |
| [7688](http://example.com/document/7688) |  	Invoice 2021 (100 EUR) |

**Reply by tlwt** [reply]
Hi @Rooyca, I just checked the docs and did not find anything. Is this possible within the current release or on the roadmap? Thanks, Till

**Reply by Rooyca** [reply]
Hello @tlwt,
I remember looking into this when you proposed it but I don't remember exactly why it was impossible to implement.
